pkg/controller/validate/validate.go

```diff
@@ -189,6 +204,76 @@ func instanceSizeForAdvancedDeployment(replicationSpecs []*mdbv1.AdvancedReplica
 	return nil
 }
 
+func instanceSizeRangeForAdvancedDeployment(replicationSpecs []*mdbv1.AdvancedReplicationSpec) error {
+	var err error
+	for _, replicationSpec := range replicationSpecs {
+		for _, regionSpec := range replicationSpec.RegionConfigs {
+			if regionSpec.AutoScaling == nil || regionSpec.AutoScaling.Compute == nil || regionSpec.AutoScaling.Compute.Enabled == nil || !*regionSpec.AutoScaling.Compute.Enabled {
```

Comment:
nitpick, for this complex logic statement it helps readability to create a function with a descriptive name that summarizes the semantics of the change.

```suggestion
			if autoscalingComputeEnabled(regionSpec.AutoScaling) {
```